### PR TITLE
Fix Required Maps after UI Review

### PR DESCRIPTION
**Pull request by nazar-kutz**

### CODE REVIEW

**Comment by vshcherb** [comment]
1.  Too many changes for 4.7 and there is no need as bugs are fixed.
2. Can't review backend code - too many changes  including formatting

- For UI fix and parameters it's possible to create separate request

File: OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java [hunk 1]
```diff
@@ -1408,4 +1413,8 @@ private void pringGC(final RoutingContext ctx, boolean before) {
 		}
 	}
 
+	public static MissingMapsCalculator getMissingMapsCalculator() {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java:1416]: static

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 2]
```diff
@@ -251,15 +272,17 @@ private boolean addMapEditions(Map<String, RegisteredMap> knownMaps, Point pnt)
 		return hhEditionPresent;
 	}
 
-	private void split(RoutingContext ctx, Map<String, RegisteredMap> knownMaps, List<Point> pointsToCheck, LatLon pnt, LatLon next) throws IOException {
+	private void split(Map<String, RegisteredMap> knownMaps, List<Point> pointsToCheck, LatLon pnt, LatLon next) throws IOException {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:275]: not tested code

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 3]
```diff
@@ -147,60 +119,109 @@ public boolean checkIfThereAreMissingMaps(RoutingContext ctx, LatLon start, List
...
+		result.potentiallyUsedMaps = convert(usedMaps);
 
-		log.info(String.format("Check missing maps %d points %.2f sec", pointsToCheck.size(),
+		LOG.info(String.format("Check missing maps %d points %.2f sec", pointsToCheck.size(),
 				(System.nanoTime() - tm) / 1e9));
-		return true;
+		return result;
 	}
 
-	private LatLon testLatLons(List<LatLon> targets) throws IOException {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:171]: code disappeared?

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 4]
```diff
@@ -147,60 +119,109 @@ public boolean checkIfThereAreMissingMaps(RoutingContext ctx, LatLon start, List
...
-		BufferedReader r = new BufferedReader(new InputStreamReader(MissingMapsCalculator.class.getResourceAsStream("/latlons.test.txt")));
-		targets.clear();
-		String s = null;
-		while ((s = r.readLine()) != null) {
-			String[] ls = s.split(",");
-			targets.add(new LatLon(Double.parseDouble(ls[1].trim()), Double.parseDouble(ls[0].trim())));
+	private Map<String, RegisteredMap> collectKnownMaps(RoutingContext ctx, String profile) {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:146]: refactoring

**Comment by nazar-kutz** [comment]
OK, I'll try to move other fixes and certain features from the code design into a new request, but keeping the necessary logic of previous fix

**Pull request closed without merging**